File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestArrayOptTimestampValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.ArrayOptTimestampInValue.ArrayOptTimestampValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestArrayOptTimestampValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestArrayOptTimestampValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                org.joda.time.DateTime[].class,
                ArrayOptTimestampValue.class.getDeclaredField("arrayOptTimestamp").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                org.joda.time.DateTime[].class,
                ArrayOptTimestampValue.class.getMethod("getArrayOptTimestamp").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                ArrayOptTimestampValue.class,
                ArrayOptTimestampValue.class.getMethod("setArrayOptTimestamp", org.joda.time.DateTime[].class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertArrayEquals(
                new org.joda.time.DateTime[0],
                new ArrayOptTimestampValue().getArrayOptTimestamp());
    }
}

File: legacy/workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestArrayOptTimestampValue.scala
package com.dslplatform.ocd.values

import ArrayOptTimestampInValue.ArrayOptTimestampValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestArrayOptTimestampValue extends SpecExtensions {
  def `Test property field type: Array[Option[org.joda.time.DateTime]]`() =
    checkType(
      typeOf[Array[Option[org.joda.time.DateTime]]]
    , typeOf[ArrayOptTimestampValue].member("arrayOptTimestamp": TermName).asMethod.returnType
    )

  def `Test property default value: Array.empty[Option[org.joda.time.DateTime]]`() {
    assert(ArrayOptTimestampValue().arrayOptTimestamp === Array.empty[Option[org.joda.time.DateTime]])
  }
}
